### ISSUE TITLE
Type unification error when using a `forall` inside a row type

### ISSUE THREAD

**Issue by matthew-healy** [tweag/nickel#1027]
**Describe the bug**

This program:

```
let r : { id : forall a. a -> a } = { id = fun r => r } in
r.id "x"
```

currently fails with an "incompatible rows declaration" error, failing to unify `forall a. a -> a` with `_a -> _a` (where `_a` is a unification variable).

**Expected behavior**

These types should successfully unify, and the code should typecheck.

**Comment by yannham** [comment]
I think this is an instance of, or at least closely related to #360, and would be fixed by closing #584.

**Comment by yannham** [comment]
Closed probably by #1473. The issue is anyway not present anymore in current 1.2.1.
